File: app/src/main/java/com/systemvx/androiddevtest/HttpUtil.java
package com.systemvx.androiddevtest;

import android.app.DownloadManager;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.Executor;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;
import java.util.concurrent.FutureTask;

import okhttp3.Call;
import okhttp3.Cookie;
import okhttp3.CookieJar;
import okhttp3.HttpUrl;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class HttpUtil {
    public static final String BASE_URL="";
    private static Map<String, List<Cookie>>cookieStore= new HashMap<>()
    private static ExecutorService thredPool= Executors.newFixedThreadPool(30);
    private static OkHttpClient okHttpClient= new OkHttpClient.Builder().cookieJar(new CookieJar()
    {
        @Override
        public void saveFromResponse(@NonNull HttpUrl httpUrl,
                                     @NonNull List<Cookie>list)
        {cookieStore.put(httpUrl.host(),list);
        }
        @Override
        public List<Cookie> loadForRequest(@NonNull HttpUrl httpUrl)
        {
            List<Cookie> cookies=cookieStore.get(httpUrl.host());
            return cookies ==null ? new ArrayList<>():cookies;
        }
    }).build();
    public static <Call> String getRequest(String url) throws Exception
    {
        FutureTask<String> task =new FutureTask<>(()->{
            DownloadManager.Request request=new Request.Builder()
                    .url(url)
                    .build();
            Call call =okHttpClient.newCall(request);
            Response response=call.execute();
            if (response.isSuccessful()&& response.body()!=null)
            {
                return response.body().string().trim();
            }
            else
            {
                return null;
            }
        });
        threadPool.submit(task);
        return task.get();
    }
    public static String postRequest(String url,
                                     Map<String,String> rawParams) throws Exception
    {
        FutureTask<String>task=new FutureTask<>(()->{
            FormBody.Builder builder=new Formbody.Builder();
            rawParams.forEach(builder::add);
            FormBody body= builder.build();
            Request request=new Request.Builder()
                    .url(url)
                    .post(body).build();
            Call call=okHttpClient.newCall(request);
            Response response=call.excute();
            if(response.isSuccessful()&& response.body!=null)
            {
                return response.body().string().trim()
            }
            else
            {
                return null;
            }
        });
        thredPool.submit(task);
        return task.get();
    }
}

File: app/src/main/java/com/systemvx/androiddevtest/HttpUtil.kt
package com.systemvx.androiddevtest

import okhttp3.*
import java.util.*
import java.util.concurrent.Callable
import java.util.concurrent.Executors
import java.util.concurrent.FutureTask

class HttpUtil {
    companion object {
        //安卓AVD虚拟机指定的宿主机回环地址
        const val BASE_URL = "http://10.0.2.2:8080"
    }


    private val cookieStore: MutableMap<String, List<Cookie>> = HashMap()
    private val threadPool = Executors.newFixedThreadPool(30)
    private val okHttpClient: OkHttpClient = OkHttpClient.Builder().cookieJar(object : CookieJar {
        override fun saveFromResponse(url: HttpUrl,
                                      cookies: List<Cookie>) {
            cookieStore[url.host] = cookies
        }

        override fun loadForRequest(url: HttpUrl): List<Cookie> {
            val cookies = cookieStore[url.host]
            return cookies ?: ArrayList()
        }
    }).build()

    fun getRequest(url: String): String? {
        val task = FutureTask(Callable<String> {
            val request: Request = Request.Builder().url(url).build()
            val call = okHttpClient.newCall(request)
            val response = call.execute()
            if (response.isSuccessful && response.body != null) {
                return@Callable response.body!!.string().trim { it <= ' ' }
            } else {
                return@Callable null
            }
        })
        threadPool.submit(task)
        return task.get()
    }

    fun postRequest(url: String,
                    rawParams: HashMap<String, String>): String {
        val task = FutureTask(Callable<String> {
            val builder = FormBody.Builder()
            rawParams.forEach { (name: String?, value: String?) -> builder.add(name, value) }
            val body: FormBody = builder.build()
            val request: Request = Request.Builder()
                    .url(url)
                    .post(body).build()
            val call = okHttpClient.newCall(request)
            val response = call.execute()
            if (response.isSuccessful && response.body != null) {
                return@Callable response.body!!.string().trim { it <= ' ' }
            } else {
                return@Callable null
            }
        })
        threadPool.submit(task)
        return task.get()
    }
}